keras_tuner/engine/base_tuner.py

```diff
@@ -170,38 +171,29 @@ def search(self, *fit_args, **fit_kwargs):
                 continue
 
             self.on_trial_begin(trial)
-            self.run_trial(trial, *fit_args, **fit_kwargs)
+            results = self.run_trial(trial, *fit_args, **fit_kwargs)
+            # `results` is None indicates user updated oracle in `run_trial()`.
+            if results is None:
+                warnings.warn(
+                    "`Tuner.run_trial()` returned None. It should return one of "
+                    "float, dict, keras.callbacks.History, or a list of one "
+                    "of these types. The use case of calling "
+                    "`Tuner.oracle.update_trial()` in `Tuner.run_trial()` is "
+                    "deprecated, which will be removed in the future.",
+                    DeprecationWarning,
```

Comment:
Add `stacklevel=2` so that users can see where the warning is coming from in their code